src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -1196,5 +1199,224 @@ internal static bool TryGetInt(FormulaValue value, out int outputValue)
+                {
+                    _flags = null;
+                    return;
+                }
+            }
+
+#pragma warning disable CS0618 // Use of [Obsolete] member: Date[Time]Value.Value
+
+            internal string JsonInternal(FormulaValue arg0, int indent)
+            {
+                return arg0 switch
+                {
+                    BlankValue blkv => "null",
+                    BooleanValue bv => bv.Value.ToString(CultureInfo.InvariantCulture).ToLowerInvariant(),
+                    ColorValue cv => GetColorString(cv.Value),
+                    DateTimeValue dtv => WithQuotes(TimeZoneInfo.ConvertTimeToUtc(dtv.Value, _timeZoneInfo).ToString("yyyy-MM-dd'T'HH:mm:ss.fffK", CultureInfo.InvariantCulture)),
+                    DateValue dv => WithQuotes(TimeZoneInfo.ConvertTimeToUtc(dv.Value, _timeZoneInfo).ToString("yyyy-MM-dd", CultureInfo.InvariantCulture)),
+                    DecimalValue dmv => dmv.Value.ToString(CultureInfo.InvariantCulture),
+                    GuidValue gv => WithQuotes(gv.Value.ToString("D", CultureInfo.InvariantCulture)),
+                    NumberValue nv => nv.Value.ToString(CultureInfo.InvariantCulture),
+                    OptionSetValue osv => GetOptionSetValueString(osv),
+                    RecordValue rv => GetJsonStringForRecord(rv, indent),
+                    StringValue sv => WithQuotes(sv.Value),
+                    TableValue tv => GetJsonStringForTable(tv, indent),
+                    TimeValue tmv => WithQuotes(tmv.Value.ToString(@"hh\:mm\:ss\.fff", CultureInfo.InvariantCulture)),
+                    _ => "invalid",
+                };
+            }
+
+#pragma warning restore CS0618
+
+            private static string WithQuotes(string str) => $@"""{str}""";
+
+            internal static string GetColorString(Color color) => WithQuotes($"#{color.R:x2}{color.G:x2}{color.B:x2}{color.A:x2}");
+
+            internal static string GetOptionSetValueString(OptionSetValue optionSetValue)
+            {
+                if (optionSetValue.Type._type.OptionSetInfo is EnumSymbol enumSymbol)
+                {
+                    if (enumSymbol.EntityName == "Color")
+                    {
+                        Color color = Color.FromArgb((int)(uint)(double)optionSetValue.ExecutionValue);
+                        return GetColorString(color);
+                    }
+
+                    return WithQuotes(optionSetValue.ExecutionValue.ToString());
+                }
+                else if (optionSetValue.Type._type.OptionSetInfo is IExternalOptionSet ops)
+                {
+                    return WithQuotes(optionSetValue.Option);
+                }
+
+                return null;
+            }
+
+            internal string GetJsonStringForRecord(RecordValue recordValue, int indent)
+            {
+                StringBuilder stringBuilder = new StringBuilder();

```

Comment:
>stringBuilder [](http://example.com/codeflow?start=30&length=13)

Instead of creating a new StringBuilder every time we go into a record / table, it is more efficient to create a single SB at the beginning (in JsonProcessing.Process) and pass it along the calls.

Reply:
Using Utf8JsonWriter object now and passing it along the various calls.